test/e2e_federation/upgrades/generic.go

```diff
@@ -58,8 +59,13 @@ func (t *GenericUpgradeTest) Teardown(f *fedframework.Framework) {
 	// Rely on the namespace deletion to clean up everything
 }
 
-// SecretUpgradeTest validates that a secret remains propagated before
-// and after a controlplane upgrade
-func SecretUpgradeTest() Test {
-	return &GenericUpgradeTest{adapter: &typeadapters.SecretAdapter{}}
+func GenericUpgradeTests() []Test {
```

Comment:
Why are docs required for this?  I get that golang religion is 'doc all the things that are exported!' but this is a test-only function, and only used once.